components/fuse/src/main/java/org/datacleaner/components/fuse/FuseStreamsComponent.java

```diff
@@ -0,0 +1,136 @@
+import org.datacleaner.api.InputRow;
+import org.datacleaner.api.MultiStreamComponent;
+import org.datacleaner.api.OutputDataStream;
+import org.datacleaner.api.OutputRowCollector;
+import org.datacleaner.components.categories.CompositionCategory;
+import org.datacleaner.job.output.OutputDataStreamBuilder;
+import org.datacleaner.job.output.OutputDataStreams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Named("Union")
+@Description("Lets you combine multiple streams into one. Providing what is equivalent to a union of tables.\n\n"
+        + "Use it to fuse data streams coming from different filter requirements, different output data streams or different source tables. "
+        + "You can define new fields whose values represent whatever is available from one of the input streams.")
+@Categorized(CompositionCategory.class)
+public class FuseStreamsComponent extends MultiStreamComponent {
+
+    private static final Logger logger = LoggerFactory.getLogger(FuseStreamsComponent.class);
+
+    public static final String OUTPUT_DATA_STREAM_NAME = "output";
+    public static final String PROPERTY_INPUTS = "Inputs";
+    public static final String PROPERTY_UNITS = "Units";
+
+    @Configured(PROPERTY_INPUTS)
+    InputColumn<?>[] _inputs;
+
+    @Configured(PROPERTY_UNITS)
+    CoalesceUnit[] _units;
+
+    private OutputRowCollector _outputRowCollector;
+    private CoalesceFunction _coalesceFunction;
+
+    public FuseStreamsComponent() {
+    }
+
+    public FuseStreamsComponent(CoalesceUnit... units) {
+        this();
+        this._units = units;
+    }
+
+    @Initialize
+    public void init() {
+        _coalesceFunction = new CoalesceFunction(false);
+    }
+
+    /**
+     * Configures the transformer using the coalesce units provided
+     * 
+     * @param units
+     */
+    public void configureUsingCoalesceUnits(CoalesceUnit... units) {
+        final List<InputColumn<?>> input = new ArrayList<InputColumn<?>>();
+        for (CoalesceUnit coalesceUnit : units) {
+            final InputColumn<?>[] inputColumns = coalesceUnit.getInputColumns(null);
+            for (final InputColumn<?> inputColumn : inputColumns) {
+                input.add(inputColumn);
+            }
+        }
+
+        _inputs = input.toArray(new InputColumn[input.size()]);
+        _units = units;
+    }
+
+    @Override
+    public void run(InputRow inputRow) {
+        final Object[] output = new Object[_units.length];
+        for (int i = 0; i < _units.length; i++) {
+            final CoalesceUnit unit = _units[i];
+            final InputColumn<?>[] inputColumns = unit.getInputColumns(_inputs);
+            final List<Object> values = inputRow.getValues(inputColumns);
+            final Object value = _coalesceFunction.coalesce(values);
+            output[i] = value;
+        }
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Fused values for row: {}", Arrays.toString(output));
+        }
```

Comment:
No need for the isDebugEnabled(). Just pass the array without toString, that is done by the logger, but only if it needs to output it.